src/main/kotlin/CorrectStopwatch.kt

```diff
@@ -0,0 +1,57 @@
+import io.reactivex.Observable
+import io.reactivex.ObservableTransformer
+import io.reactivex.Scheduler
+import io.reactivex.functions.BiFunction
+import io.reactivex.observers.TestObserver
+import io.reactivex.schedulers.TestScheduler
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.PublishSubject
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicLong
+
+
+interface CorrectStopwatch {
+
+    val value: Observable<Long>
+    fun pause()
+    fun resume()
+
+    class Impl(private val scheduler: Scheduler) : CorrectStopwatch {
+
+        override val value: Observable<Long>
+            get() = combinedParts
+
+        // we need this to control combineLatest so that it only emits when valueEmitterThing emits, not when manager emits
+        private fun valueChecker(valueEmitterThingValue : Long, managerValue : Long) : Long {
+            val newValue : Long
+            if (valueEmitterThingCounter.get() - 1 != valueEmitterThingKeeper.get())
+                newValue = (valueEmitterThingValue + 1) * managerValue
+            else
+                newValue = 0
+            valueEmitterThingKeeper.set(valueEmitterThingCounter.get() - 1)
+            return newValue
+        }
+
+        private val actualValueCounter = AtomicLong()
+        private val valueEmitterThingCounter = AtomicLong()
+        private val valueEmitterThingKeeper = AtomicLong(200)
+        private val manager = PublishSubject.create<Long>()
+        private val valueEmitterThing = BehaviorSubject.interval(1, TimeUnit.SECONDS, scheduler)
+        private val combinedParts = Observable.combineLatest(valueEmitterThing
+                    .map{ _ -> valueEmitterThingCounter.getAndIncrement()},
+                manager.scan{ x, y -> x + y },
+                BiFunction {valueEmitterThingValue : Long, managerValue : Long
+                    -> valueChecker(valueEmitterThingValue, managerValue)})
+            .filter{ x -> x > 0 }
+            .map{ _ -> actualValueCounter.getAndIncrement() }
+            .share()
+
+        override fun pause() {
+            manager.onNext(-1)
+        }
+
+        override fun resume() {
+            manager.onNext(1)
+        }
+    }
```

Comment:
It’s a good start, but can be simplified.

* Rename `manager` to `consumersCountChange`. Also, make it serialized.
* Change `value` to be `BehaviorSubject`. Also serialized.
* Create an `init` block. In it subscribe to `consumersCountChange`, `scan` from it. In `scan` you’ll be able to track current consumers count. Based on it either `switchMap` to `Observable.timer`, `Observable.never` or do nothing. This stream should subscribe directly to `value`.

This way it will be possible to have a single chain of command.